app/src/components/form/LModal.vue

```diff
@@ -0,0 +1,28 @@
+<script setup lang="ts">
+type Props = {
+    isVisible: boolean;
+    heading: string;
+};
+defineProps<Props>();
+
+const emit = defineEmits(["close"]);
+</script>
+
+<template>
+    <Teleport to="body">
+        <div v-if="isVisible" @click.self="emit('close')">
+            <div class="fixed inset-0 z-50 bg-slate-800 bg-opacity-50 backdrop-blur-sm"></div>
+            <div class="fixed inset-0 z-50 flex items-center justify-center rounded-lg p-2">
+                <div class="w-full max-w-md rounded-lg bg-white p-5 shadow-xl dark:bg-slate-700">
+                    <h2 class="mb-4 text-lg font-semibold">{{ heading }}</h2>
+                    <div class="divide-y divide-zinc-200 dark:divide-slate-600">
+                        <slot name="header"></slot>
+                    </div>
+                    <div class="mt-4">
+                        <slot name="footer">Default content for footer</slot>
```

Comment:
Shall we leave this empty (i.e. no default text). If a developer does not want to use the footer, he needs to add content to it at the moment to override the default text (as far as I can see how it will work...)